### DISCUSSION TITLE
Add typing to locale on defineConfig

### DISCUSSION THREAD

**Discussion by matheus-silva-fator3** [vueform/vueform#181] [Ideas]
### TLDR: When the defineConfig receives an invalid localle, the framework still works but fails randomly whenever it tries to access the translation feature of some components with an unclear error message.

```ts
export default defineConfig({
  theme: tailwind,
  locales: { ptBr },
  locale: "pt_BR",
});
```

I made the mistake of writing `pt_BR` instead of the correct `ptBr` on the config file. Everything seemed fine, so I kept working until I tried to use the ListElement, and it started randomly failing with the following error:

> Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'elements')

As you can see, it doesn't mention anything about translation, and I spent half an hour trying to understand what the problem was. 

So I would like to ask to throw a configuration error or at least add a typescript typing so we can Ctrl + Space the correct locale without any silly mistakes since it is not typed as a string | null.

I would love to help solve this problem, but I don't know how to do it since it would be my first time contributing to a repo.

```ts
export interface VueformConfig {
// ...
  locale: null | string;
}
```
